hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneKey.java

```diff
@@ -70,11 +70,19 @@ public class OzoneKey {
    * Constructs OzoneKey from OmKeyInfo.
    *
    */
+
+  /**
+   * The update ID of an existing key. This will be null if this OzoneKey
+   * object has not been created from an existing key read from OM, as OM allocates
+   * the update ID on commit of the key.
+   */
+  private final Long updateID;
```

Comment:
We are leaking internal names into public APIs. For public facing API, I would model it how google APIs have documented theirs : https://cloud.google.com/storage/docs/metadata#generation-number
 (We do not have the split between metadata only updates vs. whole object.)

For now, I would change this to `generation` and document the guarantees that an application has around the use of generation.  This should apply to the entire PR, where we use `updateID` which is a more nebulous name.